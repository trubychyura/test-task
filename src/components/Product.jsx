import { connect } from 'react-redux';
import { showModal } from '../redux/modal';

const Product = ({ showModal, data }) => {
  const handleClick = () => {
    showModal(data);
  };

  return (
    <tr onClick={handleClick}>
      <td>{data.name}</td>
    </tr>
  );
};

export default connect(() => ({}), { showModal })(Product);
