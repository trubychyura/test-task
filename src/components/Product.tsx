import { FC } from 'react';
import { Link } from 'react-router-dom';

import { ProductProps } from '../types';

const Product: FC<ProductProps> = ({ id, name }) => (
  <tr>
    <td>
      <Link to={`/${id}`}>{name}</Link>
    </td>
  </tr>
);

export default Product;
