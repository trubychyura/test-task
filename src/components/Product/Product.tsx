import { FC } from 'react';
import { Link } from 'react-router-dom';

import { ProductProps } from '../../types';
import { StyledCol, StyledImage, StyledRow } from '../../styled/StyledProduct';

const Product: FC<ProductProps> = ({ id, name, img }) => (
  <StyledRow>
    <StyledCol xs={{ span: 12, order: 'last' }} md={{ order: 'first' }}>
      <Link to={`/${id}`}>{name}</Link>
    </StyledCol>
    <StyledCol xs={{ span: 12, order: 'first' }} md={{ order: 'last' }}>
      <StyledImage src={img} />
    </StyledCol>
  </StyledRow>
);

export default Product;
