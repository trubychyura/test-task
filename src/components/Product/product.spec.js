import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import Product from './Product';

const setUp = (props) => shallow(<Product {...props} />);

describe('Product component', () => {
  let component;
  beforeEach(() => {
    component = setUp({ id: 'mock id' });
  });

  it('Link should contain prover text', () => {
    const name = 'mock text';
    const component = setUp({ name });
    const wrapper = component.find(Link);
    expect(wrapper.text()).toEqual(name);
  });

  it('should render Product component', () => {
    expect(component).toMatchSnapshot();
  });
});
