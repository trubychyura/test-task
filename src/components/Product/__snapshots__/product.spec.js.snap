// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render Product component 1`] = `
<Styled(Row)
  className="product"
>
  <Styled(Col)>
    <Link
      to="/mock id"
    />
  </Styled(Col)>
  <Styled(Col)>
    <Styled(Image) />
  </Styled(Col)>
</Styled(Row)>
`;
